### ISSUE TITLE
ERROR: key not found: "HOME"

### ISSUE THREAD

**Issue by marmalodak** [JuliaLang/IJulia.jl#309]
My new ipython notebook seems to work but I get this from my julia kernel.  How can I fix this?

```
[I 00:41:00.526 NotebookApp] Kernel started: 3e82deed-4a53-4cb4-9d08-8178f1aa13f2
[I 00:41:19.506 NotebookApp] Kernel shutdown: 3e82deed-4a53-4cb4-9d08-8178f1aa13f2
[I 00:41:28.343 NotebookApp] Creating new notebook in
[I 00:41:29.629 NotebookApp] Kernel started: 526e9613-5f6c-4dc2-9863-726e6c8ce928
[W 00:41:29.818 NotebookApp] 404 GET /ipython/kernelspecs/julia-0.3/logo-64x64.png (10.161.2.175) 145.61ms referer=http://qa-ipython.tintri.com/ipython/notebooks/Untitled1.ipynb?kernel_name=julia-0.3
ERROR: key not found: "HOME"
in load_juliarc at /usr/bin/../lib64/julia/sys.so
in process_options at /usr/bin/../lib64/julia/sys.so
in _start at /usr/bin/../lib64/julia/sys.so
```

**Comment by stevengj** [comment]
Sounds like you have a bug in your `.juliarc.jl` file?  (This is a file in your home directory with user code, i.e. your code, that is loaded when Julia is launched.)

**Comment by stevengj** [comment]
(What operating system and Julia version are you running?)

**Comment by marmalodak** [comment]
I'm using Fedora 21.  I've created a virtualenv for ipython3 which seems to work well.  However, I've just learned that I might have to use the jupyter/jupyterhub packages rather than the ipython packages?  I'll try that.

**Comment by stevengj** [comment]
On a Unix system, the `Base.homedir()` command assumes that you have a `HOME` environment variable...is there any reason why this might not be defined in a virtualenv?

**Comment by marmalodak** [comment]
It is defined and it points to /root.  At least, I see a $HOME when I manually activate the venv and look.

**Comment by stevengj** [comment]
That's weird, because the `ERROR: key not found: "HOME"` would be the error that would occur if `ENV["HOME"]` (which is run to find the home directory when loading the juliarc file) failed to find that environment variable.

**Comment by stevengj** [comment]
Closed by JuliaLang/julia#19636
